xcode/Subconscious/Shared/Components/Common/Byline/PetnameView.swift

```diff
@@ -7,54 +7,111 @@
 
 import SwiftUI
 
+public enum NameVariant {
+    case known(Slashlink, Petname.Name)
+    case named(Slashlink, Petname.Name)
+    case unknown(Slashlink)
+}
+
+struct PeerView: View {
+    var peer: Peer
+
+    var body: some View {
+        HStack(spacing: AppTheme.unit) {
+            ZStack {
+                RoundedRectangle(cornerRadius: AppTheme.cornerRadius/2)
+                    .foregroundColor(.secondaryBackground)
+                Text("AKA")
+                    .foregroundColor(.secondary)
+                    .font(.system(size: 10))
+            }
+            .frame(width: 28, height: 15)
+
+            Text(peer.markup)
+                .foregroundColor(.secondary)
+                .fontWeight(.regular)
+                .font(.caption)
+        }
+    }
+}
+
 /// Byline style for displaying a petname
 struct PetnameView: View {
-    var address: Slashlink
-    var name: Petname?
+    var name: NameVariant
+    var annotation: String? = nil
+    
+    public func annotated(with annotation: String) -> Self {
+        var this = self
+        this.annotation = annotation
+        return this
+    }
     
     var body: some View {
-        if let name = name {
-            VStack(alignment: .leading) {
-                Text(name.description)
-                if let peer = address.peer?.markup {
-                    Text(peer)
-                        .foregroundColor(.secondary)
-                        .fontWeight(.regular)
-                        .font(.caption)
+        VStack(alignment: .leading, spacing: AppTheme.unit) {
+            switch name {
+            case .known(let address, let name):
+                Text(name.toPetname().markup)
+                    .fontWeight(.medium)
+                    .foregroundColor(.accentColor)
+              
+                /// Prevent showing _exactly_ the same name above and below e.g.
+                /// @bob
+                /// AKA @bob
+                ///
+                /// We still permit:
+                /// @bob
+                /// AKA @bob.alice
+                if let peer = address.peer,
+                   let petname = address.petname,
+                   petname != name.toPetname() {
+                    PeerView(peer: peer)
                 }
+            case .named(let address, let name):
+                Text("\(annotation ?? "")\(name.description)")
+                    .italic()
+                    .fontWeight(.medium)
+                if let peer = address.peer {
+                    PeerView(peer: peer)
+                }
+            case .unknown(let address):
+                Text(address.peer?.markup ?? address.markup)
+                    .lineLimit(1)
+                    .fontWeight(.medium)
```

Comment:
I think we want to treat this case (other-proposed name) the same as the self-proposed name case above.

```
Maybe: bob
AKA @bob.alice
```

We don't have a self-proposed name from bob, so we use the name that alice proposes instead. This is a valid fallback.

In the case the address does not have a `Peer.petname`, we could fall back on just showing the first N characters of the did.